Linux/Sources/ContextAgent.cpp

```diff
@@ -83,11 +85,15 @@ std::tuple<std::string, std::string> ContextAgent::getWindowInfo() {
 void ContextAgent::Run(std::atomic<bool> &sigReceived) {
     _sigReceived = sigReceived.load();
     setlocale(LC_ALL, "");
-    _display = XOpenDisplay(nullptr);
-    if (_display == nullptr) {
+    _display = std::unique_ptr<Display, std::function<void(Display * )>>(XOpenDisplay(nullptr), [](Display *ptr) {
+        if (ptr != nullptr) {
+            XCloseDisplay(ptr);
```

Comment:
Seems wrong to free a pointer managed via a `unique_pointer`. What are you trying to do?